src/config.rs

```diff
@@ -129,59 +129,98 @@ impl Config {
                     slot.dbtype = Some(typ_.to_string());
                     slot.dbpath = Some(name.to_string());
                     /* if this fails there will be no slots defined */
-                    let _ = conf.add_slot(slot);
+                    let _ = conf.slots.push(slot);
                 }
                 Err(_) => (),
             }
         }
-        return Ok(conf);
+        Ok(conf)
+    }
+
+    fn fix_slot_numbers(&mut self) {
+        let mut slotnum: u32 = 0;
+        /* if there are any slot missing a valid slot number
+         * we are going to allocate slots numbers after the highest
+         * properly configured one. Note that the config file format
+         * requires slot numbers, so this generally happens for legacy
+         * or init args configurations only, ie a single slot */
+        let mut missing = false;
+        for slot in &self.slots {
+            if slot.slot != u32::MAX {
+                if slotnum <= slot.slot {
```

Comment:
just a nit that this condition is awkward to read and I would naturally like to read the condition the other way round as, where I compare the "iterator" variable to some other value, not the other way round:
```
                if slot.slot >= slotnum  {
```
This also sounds to me like the test `if (0 == function(...))`.

Reply:
My reading is reversed, here the "subject" is slotnum, and I am comparing it to the many slot.slot that we are parsing through to increment it as needed ...